CHANGELOG.md

```diff
@@ -97,6 +97,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 - Fix typo super->sb in method toString() of RemoteStoreNodeAttribute ([#15362](https://github.com/opensearch-project/OpenSearch/pull/15362))
 - [Workload Management] Fixing Create/Update QueryGroup TransportActions to execute from non-cluster manager nodes ([16422](https://github.com/opensearch-project/OpenSearch/pull/16422))
 - Fix flaky test in `testApproximateRangeWithSizeOverDefault` by adjusting totalHits assertion logic ([#16434](https://github.com/opensearch-project/OpenSearch/pull/16434#pullrequestreview-2386999409))
+- Fix string equality comparisons ([#16461](https://github.com/opensearch-project/OpenSearch/pull/16461))
```

Comment:
we dont need changelog for such changes

Reply:
Can we have it merged as is? I don't fancy creating one more PR after another unsuccessful rebase

Reply:
There's already a conflict in CHANGELOG, gradle checks failed, and the diff looks big now :( Appreciate your help in rebasing this.